src/Ilios/AuthenticationBundle/RelationshipVoter/CurriculumInventoryInstitution.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Ilios\AuthenticationBundle\RelationshipVoter;
+
+use Ilios\AuthenticationBundle\Classes\SessionUserInterface;
+use Ilios\CoreBundle\Entity\CurriculumInventoryInstitutionInterface;
+use Ilios\CoreBundle\Entity\DTO\CurriculumInventoryInstitutionDTO;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+class CurriculumInventoryInstitution extends AbstractVoter
+{
+    protected function supports($attribute, $subject)
+    {
+        return (
+            ($subject instanceof CurriculumInventoryInstitutionDTO && in_array($attribute, [self::VIEW])) or
+            ($subject instanceof CurriculumInventoryInstitutionInterface && in_array($attribute, [
+                    self::CREATE, self::VIEW, self::EDIT, self::DELETE
+                ]))
+        );
+    }
+
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+        if (!$user instanceof SessionUserInterface) {
+            return false;
+        }
+        if ($user->isRoot()) {
+            return true;
+        }
+
+        if ($subject instanceof CurriculumInventoryInstitutionDTO) {
+            return $this->voteOnDTO($user, $subject);
+        }
+
+        if ($subject instanceof CurriculumInventoryInstitutionInterface) {
+            return $this->voteOnEntity($attribute, $user, $subject);
+        }
+
+        return false;
+    }
+
+    protected function voteOnDTO(
+        SessionUserInterface $sessionUser,
+        CurriculumInventoryInstitutionDTO $institution
+    ): bool {
+        return $this->permissionChecker->canReadCurriculumInventoryInstitution($sessionUser, $institution->school);
```

Comment:
This might be one of the things that any logged in user can read. I'm pretty sure the AAMC code is public information and everything else in here is for sure.

Reply:
so far, we have not considered the notion of a blanket read permission for authenticated users. if so, then we should apply the same rule for most of our data points.